server/controllers/placementApplications/reviewController.test.ts

```diff
@@ -0,0 +1,358 @@
+          ...request,
+          body: {
+            summaryOfChanges: 'some changes',
+          },
+        },
+        response,
+      )
+
+      expect(response.redirect).toHaveBeenCalledWith(
+        placementApplicationPaths.placementApplications.review.decision({ id }),
+      )
+    })
+
+    it('renders errors if the changes summary is empty', async () => {
+      const err = new Error('Invalid request body')
+      const errors = { summaryOfChanges: 'some error' }
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return {
+          errors,
+          update: jest.fn(() => {
+            throw err
+          }),
+        }
+      })
+      ;(generateErrorMessages as jest.Mock).mockReturnValue('some error')
+      ;(generateErrorSummary as jest.Mock).mockReturnValue('some error summary')
+
+      const requestHandler = reviewController.update()
+
+      await requestHandler(
+        {
+          ...request,
+          body: {
+            summaryOfChanges: '',
+          },
+        },
+        response,
+      )
+
+      expect(generateErrorMessages).toHaveBeenCalledWith(errors)
+      expect(generateErrorSummary).toHaveBeenCalledWith(errors)
+
+      expect(request.flash).toHaveBeenCalledWith('errors', 'some error')
+      expect(request.flash).toHaveBeenCalledWith('errorSummary', 'some error summary')
+      expect(request.flash).toHaveBeenCalledWith('userInput', { summaryOfChanges: '' })
+    })
+
+    it('passes unexpected error to catchValidationErrorOrPropogate', async () => {
+      const err = new Error()
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return {
+          update: jest.fn(() => {
+            throw err
+          }),
+        }
+      })
+
+      const requestHandler = reviewController.update()
+
+      await requestHandler(request, response)
+
+      expect(catchValidationErrorOrPropogate).toHaveBeenCalledWith(request, response, err, request.headers.referer)
+    })
+  })
+
+  describe('submit', () => {
+    const request: DeepMocked<Request> = createMock<Request>({
+      headers: { referer: 'some-referrer' },
+      params: { id },
+      user: { token: 'some-token' },
+      session: createMock<Request['session']>({
+        placementApplicationDecisions: { [id]: { summaryOfChanges: 'some changes' } },
+      }),
+      flash: jest.fn(),
+    })
+
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('redirects to the confirm path on success', async () => {
+      ;(PlacementApplicationReview as jest.Mock).mockImplementation(() => {
+        return {
+          applicationId: id,
+          update: jest.fn(),
+        }
+      })
+
+      const requestHandler = reviewController.submit()
+
+      await requestHandler(
+        {
+          ...request,
+          body: {
+            decision: 'a decision',
+            decisionSummary: 'a decision summary',
+          },
+        },
+        response,
+      )
+
```

Comment:
As above, we should be check that `PlacementApplicationReview.update` was called here